### PR TITLE
[BUGFIX] Have index updated when using frontend editing

### PR DESCRIPTION
**Pull request by megamisan**
DataHandler (TceMain) hooks are registered also in frontend mode when the TSFE_EDIT post parameter array is present.

Fixes: #647

### CODE REVIEW

File: ext_tables.php [hunk 1]
```diff
@@ -132,7 +132,8 @@
         'solr::clearSolrConnectionCache',
         'ApacheSolrForTypo3\\Solr\\ConnectionManager->updateConnections'
     );
-
+}
+if ((TYPO3_MODE == 'BE') || ((TYPO3_MODE == 'FE') && isset($_POST['TSFE_EDIT']))) {
```
Comment (irnnr) [inline ext_tables.php]: Can you make two changes here: 
1) You don't need the braces around `TYPO3_MODE == 'FE'`
2) Please use triple equal signs for comparison `===`

Comment (megamisan) [inline ext_tables.php]: Done.

**Comment by stmllr** [comment]
cHash generation has changed with latest TYPO3 release for new installations and now takes id into calculation.

**Comment by irnnr** [comment]
ah right, thanks @stmllr

**Comment by timohund** [comment]
I've fixed the broken tests that are related to the cHash change. Tests are running through now and since everything else looks good, i'll merge this PR.

**Pull request merged**